File: src/dotnet/Common/Models/ResourceProviders/Prompt/MultipartPrompt.cs
using FoundationaLLM.Common.Constants.ResourceProviders;
using System.Text.Json.Serialization;


namespace FoundationaLLM.Common.Models.ResourceProviders.Prompt
{
    /// <summary>
    /// Multipart prompt composed of a prefix and a suffix.
    /// </summary>
    public class MultipartPrompt : PromptBase
    {
        /// <summary>
        /// The prompt prefix.
        /// </summary>
        [JsonPropertyName("prefix")]
        public string? Prefix { get; set; }
        /// <summary>
        /// The prompt suffix.
        /// </summary>
        [JsonPropertyName("suffix")]
        public string? Suffix { get; set; }

        /// <summary>
        /// Set default property values.
        /// </summary>
        public MultipartPrompt() =>
            Type = PromptTypes.Multipart;
    }

}

File: src/python/PythonSDK/foundationallm/models/resource_providers/prompts/multipart_prompt.py
from typing import Optional
from pydantic import Field
from foundationallm.models.resource_providers import ResourceBase
from foundationallm.models.resource_providers.prompts import PromptTypes

class MultipartPrompt(ResourceBase):
    """
    Encapsulates a multipart prompt composed of a prefix and a suffix.
    """
    type: str = Field(default=PromptTypes.MULTIPART, description='The type of the prompt.')
    prefix: Optional[str] = Field(default='', description='The prefix of the prompt.')
    suffix: Optional[str] = Field(default='', description='The suffix of the prompt.')
